cmd/kflex/common/cp.go

```diff
@@ -18,10 +18,76 @@ package common
 
 import (
 	"context"
+	"fmt"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	tenancyv1alpha1 "github.com/kubestellar/kubeflex/api/v1alpha1"
+	"github.com/kubestellar/kubeflex/pkg/util"
 )
 
 type CP struct {
 	Ctx        context.Context
 	Kubeconfig string
 	Name       string
 }
+
+func GenerateControlPlane(name, controlPlaneType, backendType, hook string, hookVars []string, tokenExpirationSeconds *int64) (*tenancyv1alpha1.ControlPlane, error) {
+	cp := &tenancyv1alpha1.ControlPlane{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: tenancyv1alpha1.ControlPlaneSpec{
+			Type:    tenancyv1alpha1.ControlPlaneType(controlPlaneType),
+			Backend: tenancyv1alpha1.BackendDBType(backendType),
+		},
+	}
+	if hook != "" {
+		cp.Spec.PostCreateHook = &hook
+		var err error
+		cp.Spec.PostCreateHookVars, err = convertToMap(hookVars)
+		if err != nil {
+			return nil, err
+		}
+	}
+	if controlPlaneType == string(tenancyv1alpha1.ControlPlaneTypeExternal) {
+		cp.Spec.BootstrapSecretRef = &tenancyv1alpha1.SecretReference{
+			Name:         util.GenerateBootstrapSecretName(name),
+			Namespace:    util.SystemNamespace,
+			InClusterKey: util.KubeconfigSecretKeyInCluster,
+		}
+		if tokenExpirationSeconds != nil {
+			cp.Spec.TokenExpirationSeconds = tokenExpirationSeconds
```

Comment:
Isn't this something to do regardless of control plane type?

Reply:
goo point, updated